### PR TITLE
[JENKINS-34021] If DurableTaskStep.Execution.listener is sometimes null

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-34021](https://issues.jenkins-ci.org/browse/JENKINS-34021)

Symptomatic & diagnostic fix.

@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java [hunk 1]
```diff
@@ -114,13 +118,33 @@ protected DurableTaskStepDescriptor() {
...
+            TaskListener l = listener;
+            if (l == null) {
+                StepContext context = getContext();
+                try {
+                    l = context.get(TaskListener.class);
+                    if (l != null) {
+                        LOGGER.log(Level.WARNING, "JENKINS-34021: DurableTaskStep.Execution.listener not restored in {0}", context);
+                    } else {
+                        LOGGER.log(Level.WARNING, "JENKINS-34021: TaskListener not even available upon request in {0}", context);
+                        l = new LogTaskListener(LOGGER, Level.FINE);
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java]: This is redirecting the log messages to the file logger, right? Maybe worth noting it in the previous WARNING log.

**Comment by amuniz** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java (hunk 1 above)
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java]: Well, if this ever happens again, a dev is going to need to supply user instructions on enabling logging etc.

**Pull request merged**